pkg/controller/autoscaling/autoscalinggroup/setup.go

```diff
@@ -219,6 +239,18 @@ func isUpToDate(_ context.Context, obj *svcapitypes.AutoScalingGroup, obs *svcsd
 	return true, "", nil
 }
 
+func fieldIsEqual[T string | bool](a, b *T) bool {
+	if a != nil && b != nil {
+		if *a != *b {
+			return false
+		}
+	}
+	if a == nil || b == nil {
+		return false
+	}
+	return true
+}
```

Comment:
Have you considered https://pkg.go.dev/k8s.io/utils@v0.0.0-20230726121419-3b25d923346b/ptr#Equal ? I think it can be used instead of a custom implementation.

Reply:
PTAL again